lib/Game_Logic/gamelogic.cpp

```diff
@@ -0,0 +1,259 @@
+#include "gamelogic.h"
+
+// initialize shared variables
+int score = 0;
+bool displayItemValueBool = false;
+
+gamelogic::gamelogic() {}
+
+void gamelogic::getSeconds(long time)
+{
+    seconds = time;
+}
+
+void gamelogic::getMilliseconds(long time)
+{
+    milliSeconds = time;
+}
+
+int *gamelogic::gameTick(Item itemsArray[])
+{
+    // gametick function calculates all the game logic and returns the information to the main file
+
+    // debounce c & z button
+    Nunchuk.getState(NUNCHUK_ADDRESS);
+    if (Nunchuk.state.c_button == 0 && justChangedC)
+    {
+        justChangedC = false;
+    }
+    // move character calculations
+    moveCharacter();
+    // hook calculations
+    hookLogic(itemsArray);
+
+    // return information to main file
+    returnInformation[CHARACTER_POSITION_X] = characterPositionX;
+    returnInformation[CHARACTER_POSITION_Y] = characterPositionY;
+    returnInformation[RESET_SKY_SIDE] = resetSkySide;
+
+    returnInformation[X_BEGIN_HOOK] = xBeginHook;
+    returnInformation[Y_BEGIN_HOOK] = yBeginHook;
+    returnInformation[X_END_HOOK] = xEndHook;
+    returnInformation[Y_END_HOOK] = yEndHook;
+    returnInformation[IS_HOOK_SWINGING] = hookSwinging;
+    returnInformation[DELETE_HOOK] = deleteHook;
+
+    returnInformation[WITHDRAW_HOOK] = withdrawHook;
+    returnInformation[X_BEGIN_REMOVE_HOOK] = xBeginRemoveHook;
+    returnInformation[Y_BEGIN_REMOVE_HOOK] = yBeginRemoveHook;
+    returnInformation[X_END_REMOVE_HOOK] = xEndRemoveHook;
+    returnInformation[Y_END_REMOVE_HOOK] = yEndRemoveHook;
+
+    returnInformation[ITEM_GRABBED_BOOL] = itemGrabbedBool;
+    returnInformation[ITEM_GRABBED] = itemGrabbed;
+
+    returnInformation[SCORE_HAS_CHANGED] = scoreHasChanged;
+
+    returnInformation[ITEM_VALUE] = itemValue;
```

Comment:
you could also instead of storing stuff in local variables give a pointer to the array in main and store them directly to that making this block of code obsolete